### PR TITLE
Convert disallowed macros to an early pass lint

### PR DESCRIPTION
**Pull request by jdonszelmann**
r? @xFrednet 

I forgot one yesterday, but this one also needs to be early pass. I use AttrStorage in a similar way as FormatArgs does. It's not actually very wasteful, there aren't *that* many attributes, and every attribute used to store its own span, which outside clippy will soon not be necessary anymore.

There's one more that needs refactoring, which @GuillaumeGomez is doing

### CODE REVIEW

File: src/tools/clippy/clippy_lints/src/disallowed_macros.rs [hunk 1]
```diff
@@ -1,8 +1,8 @@
+
```
Comment (GuillaumeGomez) [inline src/tools/clippy/clippy_lints/src/disallowed_macros.rs:1]: nit :)

File: src/tools/clippy/clippy_lints/src/disallowed_macros.rs [hunk 2]
```diff
@@ -164,8 +180,5 @@ impl LateLintPass<'_> for DisallowedMacros {
     fn check_path(&mut self, cx: &LateContext<'_>, path: &Path<'_>, _: HirId) {
         self.check(cx, path.span, None);
     }
-
-    fn check_attribute(&mut self, cx: &LateContext<'_>, attr: &Attribute) {
-        self.check(cx, attr.span, self.derive_src);
-    }
 }
+
```
Comment (GuillaumeGomez) [inline src/tools/clippy/clippy_lints/src/disallowed_macros.rs:184]: Extra line here too.

**Comment by GuillaumeGomez** [comment]
I think `fmt` check will be unhappy but otherwise code looks good to me. Thanks!

**Comment by jdonszelmann** [comment]
actually I don't think so, I ran tidy on it haha

**Comment by jdonszelmann** [comment]
image
yup, but I'll force push over it fixing your comments

**Review by flip1995 — CHANGES_REQUESTED** [review]
This only modifies Clippy. Please open this PR towards the Clippy repo.

To do this easily you can do

```
git diff HEAD~1 --patch -- src/tools/clippy | sed "s|src/tools/clippy/||" > clippy.patch
```

and then in the Clippy repo clone 

```
git apply /path/to/clippy.patch
```

**Comment by GuillaumeGomez** [comment]
Oh good catch. Didn't even notice it was the wrong repository...

**Comment by flip1995** [comment]
Also the other PR should have been done to the Clippy repo.

To give an explanation for that policy: The Clippy repo runs more tests on Clippy than in the Rust repo. It also uses slightly different formatting rules. Those need to be fixed during syncing the Clippy and the Rust repo. The more changes were done between syncs to Clippy in the Rust repo, the more conflicts there are and the longer the sync takes.

The most important check that is done in Clippy for changes like this: Clippy does a mini-crater run on every PR to see the effect of changes. I have the feeling that a change like this (or rather the ones in the previous PR) can have unexpected effects. Those we (meaning I) would need to catch during the next sync.

**Comment by jdonszelmann** [comment]
Oh alright, that makes sense. I'll refile it.

**Comment by jdonszelmann** [comment]
closed in favor of https://github.com/rust-lang/rust-clippy/pull/13657

**Comment by flip1995** [comment]
Thank you. I will do the sync tomorrow, as it is already delayed by 1 week because of other changes that blocked it, that needed to be fixed.

So you only have to wait until tomorrow evening :)

**Pull request closed without merging**